packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -117,25 +93,36 @@ export const useCheckoutPage = props => {
     }, [checkoutStep, setCheckoutStep]);
 
     const handlePlaceOrder = useCallback(async () => {
-        await getOrderDetails({
-            variables: {
-                cartId
-            }
-        });
-
-        await placeOrder({
-            variables: {
-                cartId
-            }
-        });
+        try {
+            await getOrderDetails({
+                variables: {
+                    cartId
+                }
+            });
+            await placeOrder({
+                variables: {
+                    cartId
+                }
+            });
 
-        await removeCart();
+            await removeCart();
 
-        await clearCartDataFromCache(apolloClient);
+            await clearCartDataFromCache(apolloClient);
 
-        await createCart({
-            fetchCartId
-        });
+            await createCart({
+                fetchCartId
+            });
+        } catch (err) {
+            console.error(
+                'An error occurred during when placing the order',
+                err
+            );
+            setReviewOrderButtonClicked(false);
+            setCheckoutStep(CHECKOUT_STEP.PAYMENT);
+            // TODO: Delete nonce? The nonce might be expired and why the order
+            // failed. If we delete it the payment info section will render as
+            // if it was not filled, thus prompting the user to enter new info.
```

Comment:
Should I resolve this here? The checkout page is not responsible for handling the payment info/credit card component's details such as a `nonce`. Similarly, the payment info/CC component have already submitted billing info/nonce to the cart when you click "review order" and it was successful at that time.

So we have some options. In either case we should probably throw a more descriptive toast error message such as "Unable to place order. Re-enter your payment information.". The question becomes whether we reset the payment form to the initial empty state or as the done state. Resetting the form requires that we delete the nonce here in the checkout page talon but I don't think that should be the checkout page's responsibility.

Reply:
Opened an issue for this: https://jira.corp.magento.com/browse/PWA-556